resources/deployment_resource.go

```diff
@@ -59,9 +60,11 @@ func (d *Deployment) UnmarshalJSON(data []byte) error {
 			Value  constant.DeploymentStatusValue  `json:"value"`
 			Reason constant.DeploymentStatusReason `json:"reason"`
 		} `json:"status"`
-		Droplet      Droplet   `json:"droplet,omitempty"`
-		NewProcesses []Process `json:"new_processes,omitempty"`
+		Droplet      Droplet                     `json:"droplet,omitempty"`
+		NewProcesses []Process                   `json:"new_processes,omitempty"`
+		Strategy     constant.DeploymentStrategy `json:"strategy"`
```

Comment:
**thought/non-blocking** Not strictly part of this PR, but I'm not sure its wise to have an enum here for the deployment strategies/states in the CLI. If the CLI is pointed at a newer version, say with canaries, then this will throw an error. 

It's probably wiser to just allow arbitrary strings in the unmarshaling here and only use constants in the CLI when the CLI needs to understand the specific context of the Deployment.

Reply:
I concur @Samze. (Both in substance and that this shouldn't block a merge.)

There are probably lots of cases where some CLI command doesn't care about the value of this field, but it will blow regardless up at the deserialization step when we use this enum.

Zooming out even further — it would be *lovely* if we could use some machine-readable API descriptor to generate these DTOs rather than trying to keep them in sync manually. Do y'all publish, e.g., an OpenAPI spec anywhere that we could consume?

Reply:
Unfortunately there currently is no OpenAPI schema for Cloud Controller API - https://github.com/cloudfoundry/cloud_controller_ng/issues/2192